tests/suites/hup/suite.js

```diff
@@ -12,14 +12,35 @@ const { join } = require('path');
 const { homedir } = require('os');
 const Docker = require('dockerode');
 const exec = require('bluebird').promisify(require('child_process').exec);
+const Bluebird = require('bluebird');
 
 // Starts registry, uploads target image to registry
 const runRegistry = async (that, seedWithImage) => {
 	const docker = new Docker();
-	const registryImage = 'registry:2';
-
-	that.log('Pulling registry image');
-	await docker.pull(registryImage);
+	const registryImage = 'registry:2.7.1';
+
+	that.log(`Pulling registry image: ${registryImage}`);
+	const stream = await docker.pull(registryImage);
+
+	await new Bluebird((resolve, reject) => {
+		docker.modem.followProgress(
+			stream,
+			(err, output) => {
+				// onFinished
+				if (!err && output && output.length) {
+					err = output.pop().error;
+				}
+				if (err) {
+					reject(err);
+				} else {
+					resolve();
+				}
+			},
+			event => {
+				// onProgress
+			},
+		);
+	});
```

Comment:
An alternative to Bluebird here would be to install `stream-to-promise` as pointed out in this comment.
https://github.com/apocas/dockerode/issues/595#issuecomment-703743591